EIPS/eip-7069.md

```diff
@@ -0,0 +1,145 @@
+
+Besides the above, this change introduces a convenience feature of returning more detailed status codes: success (0), revert (1), failure (2). This moves from the boolean option to codes, which are extensible in the future.
+
+Lastly, the introduction of the `RETURNDATA*` instructions ([EIP-211](./eip-211.md)) have obsoleted the output parameters of calls, mostly rendering them unused. An interesting problem to mention is the case of [ERC-20](./eip-20.md) where conflicting implementations caused a lot of trouble, where some would return something, while others wouldn't.
+
+## Specification
+
+| Name | Value | Comment |
+|------|-------|---------|
+| WARM_STORAGE_READ_COST | 100 | From [EIP-2929](./eip-2929.md) |
+| COLD_ACCOUNT_ACCESS | 2600 | From [EIP-2929](./eip-2929.md) |
+| ACCOUNT_CREATION_COST | 25000 | |
+| MIN_RETAINED_GAS | 5000 | |
+| MIN_CALLEE_GAS | 2300 | |
+
+We introduce three new instructions:
+
+- `CALL2` (`0xf8`) with arguments `(target_address, input_offset, input_size, value)` 
+- `DELEGATECALL2` (`0xf9`) with arguments `(target_address, input_offset, input_size)`
+- `STATICCALL2` (`0xfb`) with arguments `(target_address, input_offset, input_size)`
+
+Execution semantics:
+
+1. Charge `WARM_STORAGE_READ_COST` (100) gas.
+2. Pop required arguments from stack, fail with error on stack underflow.
+4. If `value` is non-zero:
+    3a. Fail with error if the current frame is in `static-mode`.
+    3b. Fail with error if the balance of the current account is less than `value`.
+4. Peform (and charge for) memory expansion using `[input_offset, input_size]`.
+5. If `target_address` is not in the `warm_account_list`, charge `COLD_ACCOUNT_ACCESS - WARM_STORAGE_READ_COST` (2500) gas.
+6. If `target_address` is not in the state and the call configuration would result in account creation, charge `ACCOUNT_CREATION_COST` (25000) gas.
+    - The only such case in this EIP is if `value` is non-zero.
+7. Calculate the gas available to callee as caller's remaining gas reduced by `max(ceil(gas/64), MIN_RETAINED_GAS)` (`MIN_RETAINED_GAS` is 5000).
+8. Fail with error if the gas available to callee at this point is less than `MIN_CALLEE_GAS` (2300).
+9. Perform the call with the available gas and configuration.
+10. Push a status code on the stack:
+    11a. `0` if the call was successful.
+    11b. `1` if the call has reverted.
+    11c. `2` if the call has failed.
+11. Gas not used by the callee is returned to the caller.
+
+Note: Unlike `CALL` there is no extra charge for value bearing calls.
+
+*TODO:* Clarify which side (caller/callee) is gas deducted from and where an error originates from.
+
+*TODO:* Mention gas refunds?
+
+*TODO:* Consider option where non-calldata value transfer is not allowed, but there's a specific `TRANSFER`/`PAY` function for that. Would simplify the logic greatly.
```

Comment:
If you put these in comments, the linter will catch them for you before moving to review:

```suggestion
<!-- *TODO:* Clarify which side (caller/callee) is gas deducted from and where an error originates from. -->

<!-- *TODO:* Mention gas refunds? -->

<!-- *TODO:* Consider option where non-calldata value transfer is not allowed, but there's a specific `TRANSFER`/`PAY` function for that. Would simplify the logic greatly. -->
```